### DISCUSSION TITLE
How to login via UPN or (non-primary) smtp_address?

### DISCUSSION THREAD

**Discussion by alexhook** [ecederstrand/exchangelib#1253] [General]
Hi!

Please help me to understand how to get an Account object without using primary_smtp_address?

**Comment by ecederstrand** [comment]
You need the primary_smtp_address eventually, so the way to do this is to use autodiscover to get the actual primary_smtp_address  value.

**Reply by alexhook** [reply]
Please tell me how to find primary_smtp_address without using another email address? I'm trying to find primary_smtp_address using the autodiscover protocol using domain\username format, but it seems like it's impossible.

**Reply by ecederstrand** [reply]
After creating the `Account` with `autodiscover=True` and the alias address, the account object will have the `primary_smtp_address` set to the real primary address.

**Reply by alexhook** [reply]
At the time of creating the Account, I do not have any email address. I only have `domain\username`. If I pass this value to Account, there will be an error :

```python
ValueError: primary_smtp_address 'domain\\username' is not an email address
```

**Reply by ecederstrand** [reply]
Ok, your initial post just hinted that you had a non-primaty smtp_address.

I don't think autodiscover works without an email address at all. `Account` is what models a mailbox, so you have to have an email address of some sort to tell EWS to connect to after authenticating. It's possible the you can just use `username@your_company.tld`.
